sync/sync.go

```diff
@@ -346,6 +347,18 @@ Register:
 		return fmt.Errorf("failed to set rebuilding: true, error: %s", err.Error())
 	}
 
+	ok, err := t.isRevisionCountAndChainSame(replicaAddress, fromClient, toClient)
+	if err != nil {
+		return err
+	}
+
+	if ok {
```

Comment:
Due to this, do we miss out punching holes during preload, as there won't be reload?

Reply:
@vishnuitta yes, you are right, we may miss few holes. How about calling reloadAndVerify here ?